### PR TITLE
LPD-35622 Predefined Variables Tests

### PR DESCRIPTION
**Pull request by joshuacords**
**Story**: https://liferay.atlassian.net/browse/LPD-28678
**Subtask**: https://liferay.atlassian.net/browse/LPD-35622

**Author:** @joshuacords

### CODE REVIEW

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
ci:test:relevant

File: modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/UserSXPParameterContributorTest.java [hunk 1]
```diff
@@ -347,6 +349,13 @@ public void testEmailDomain() throws Exception {
 			"user.email_domain");
 	}
 
+	@Test
+	public void testExternalReferenceCode() throws Exception {
+		_testSXPParameter(
+			value -> value == _user.getExternalReferenceCode(),
```
Comment (gustavolimav) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/UserSXPParameterContributorTest.java:355]: `.equals`? as `userExternalReferenceCode` is a String

File: modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/ContextSXPParameterContributorTest.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+ */
+public class ContextSXPParameterContributorTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
```
Comment (gustavolimav) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/ContextSXPParameterContributorTest.java:48]: `@BeforeClass` `setUpClass()` seems better as we never change the return value of `language`.

File: modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/ContextSXPParameterContributorTest.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+			_mockGroupLocalService(), _language);
+
+		_contextSXPParameterContributor.contribute(
+			_exceptionListener, _searchContext, _sxpParameters);
+
+		Assert.assertTrue(_exists(sxpParameterName, unsafePredicate));
+	}
+
+	private ContextSXPParameterContributor _contextSXPParameterContributor;
+	private final ExceptionListener _exceptionListener = Mockito.mock(
```
Comment (gustavolimav) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/ContextSXPParameterContributorTest.java:248]: Inline we use only once

File: modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/ContextSXPParameterContributorTest.java [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+		_contextSXPParameterContributor = new ContextSXPParameterContributor(
+			_mockGroupLocalService(), _language);
+
+		_contextSXPParameterContributor.contribute(
+			_exceptionListener, _searchContext, _sxpParameters);
+
+		Assert.assertTrue(_exists(sxpParameterName, unsafePredicate));
+	}
+
+	private ContextSXPParameterContributor _contextSXPParameterContributor;
```
Comment (gustavolimav) [inline modules/dxp/apps/search-experiences/search-experiences-service/src/test/java/com/liferay/search/experiences/internal/blueprint/parameter/contributor/ContextSXPParameterContributorTest.java:247]: could be a local variable as we only use it here.

**Review by gustavolimav — CHANGES_REQUESTED** [review]
Hey @joshuacords, in general looks good, only couple chages.

**Comment by joshuacords** [comment]
ci:test:sf

**Comment by joshuacords** [comment]
ci:test:relevant

**Comment by joshuacords** [comment]
ci:test:search

**Comment by joshuacords** [comment]
Hi @gustavolimav - made those changes, thanks.

**Review by gustavolimav — APPROVED** [review]
LGTM! Thx for the chages!

**Comment by joshuacords** [comment]
ci:forward

**Pull request closed without merging**